packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -564,37 +578,90 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         }
     }
 
-    private tooltip(x: number, y: number): void {
+    private closestPoint(array: any, point: any, xRatio: number, yRatio: number, chartHeight: number): number | undefined {
+        const hypos: any[] = [];
+
+        array.forEach((ar: any) => {
+            // Convert x values to screen pixels
+            const pointX = point.x * xRatio;
+            const x = ar.x * xRatio;
+
+            // Convert y values to screen pixels
+            const pointY: number = chartHeight - point.y;
+            const y = (ar.y - this.state.allMin) * yRatio;
+
+            // Calculate hypotenuses
+            const hypotenuse = Math.sqrt(Math.pow(Math.abs(pointX - x), 2) + Math.pow(Math.abs(pointY - y), 2));
```

Comment:
If you print the hypotenuse in real time, you will see that distance 0 is not exactly where the mouse is over a point. So there is something slightly off in the calculations somewhere.

Reply:
Could it be something to do with the margins?

Is there a way for the chart to provide the exact coordinates of a data point instead of the client trying to figure it out?

Reply:
remove the square root, please, useless calculations

Reply:
You also really don't need an abs when it comes to squares. a*a  is always positive, if not, I will panic since we're dealing with imaginary points.